controllers/user.controllers.js

```diff
@@ -0,0 +1,147 @@
+const { response } = require("express")
+const config = require('../config/const')
+const flash = require('connect-flash')
+
+//importing user model
+const user = require('../models/user')
+
+//rendering to website home page
+module.exports.index = (req, res) => {
+    res.render('index',{ index:'index' })
+}
+
+//rendering to user signup page
+module.exports.signup = (req, res) => {
+    res.render('signup',{ index:'index' })
+}
+
+//function for create user through signup page 
+module.exports.signupAction = async (req, res, next) => {
+    try{
+        const { name, email, mobile ,password } = req.body;
+        const resultUser = await user.findOne({$or:[{email:email},{mobile:mobile}]});
+        if(resultUser != null) {
+            if(resultUser.email == email) {
+                res.status(config.statusCode.UNPROCESSABLE_ENTITY).render('signup',{ msg1:'email already exist' })
+            } else if(resultUser.mobile == mobile) {
+                res.status(config.statusCode.UNPROCESSABLE_ENTITY).render('signup',{ msg1:'mobile no. already exist' })
+            }
+        } else {
+            const newUser = new user()
+            newUser.name=name,
+            newUser.email=email,
+            newUser.mobile=mobile,
+            newUser.setPassword(password)
+            newUser.save((err,user) => {
+                if(err) throw err;
+                else {
+                    res.status(config.statusCode.CREATED).render('login',{msg:'registration successful', index:'index'})
+                }
+            })
+        }
+        } catch(err) {
+        console.log(err);
+    }
+}
+
+//rendering to user login page
+module.exports.login = (req, res) => {
+    res.render('login',{ index:'index' })
+}
+
+//function for login user that check user is exist in database or not if exist then login
+module.exports.loginAction = async (req, res, next) => {
+    try{
+        const { email, password } = req.body;
+        const resultUser = await user.findOne({ email:email });
+        console.log(resultUser);
+            if(resultUser != null) {
```

Comment:
No need to add check like if(resultUser != null) make it like if(resultUser) to every where when you have added check for null like this